index.bs

```diff
@@ -3199,6 +3201,143 @@ The [=remote end steps=] with |command parameters| are:
 1. Return a new map matching the <code>ScriptEvaluateResultSuccess</code>
    production, with the <code>result</code> field set to |result|.
 
+#### The script.invoke Command ####  {#command-script-invoke}
+
+The <dfn export for=commands>script.invoke</dfn> command invokes a provided
+function with given arguments in a given realm.
+
+ <code>RealmInfo</code> can be either a realm or a browsing context.
+
+Issue: declare `RemoteValueArgument`.
+
+Issue: declare `LocalValueArgument`.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      ScriptInvokeCommand = {
+        method: "script.invoke",
+        params: ScriptInvokeParameters
+      }
+
+      ScriptInvokeParameters = {
+        functionDeclaration: text;
+        arguments: [InvokeArgument];
+        awaitPromise?: bool;
+        target: Target;
+      }
+
+      InvokeArgument = RemoteValueArgument | LocalValueArgument;
+
+      // TODO: declare.
+      RemoteValueArgument = {
+        objectId: string;
+      };
+
+      // TODO: declare.
+      LocalValueArgument = {
+        type: string;
+        value: any;
+      };
+
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      ScriptInvokeResult = (
+        ScriptInvokeResultSuccess //
+        ScriptInvokeResultException
+      )
+
+      ScriptInvokeResultSuccess = {
+         result: RemoteValue
+      }
+
+      ScriptInvokeResultException = {
+        exceptionDetails: ExceptionDetails
+      }
+    </pre>
+   </dd>
+</dl>
+
+Issue: TODO: Add timeout argument as described in the script.evaluate.
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |realm| be the result of [=trying=] to [=get a realm from a target=]
+   given the value of the <code>target</code> field of |command parameters|.
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=realm execution context=]'s Realm component is |realm|.
+
+1. Let |function declaration| be the value of the <code>functionDeclaration</code> field of |command
+   parameters|.
+
+1. Let |arguments declaration| be the value of the <code>arguments</code> field of |command
+   parameters|.
+
+1. Let |await promise| be the value of the <code>awaitPromise</code> field of
+   |command parameters|, if present, or <code>true</code> otherwise.
+
+1. Let |options| be the [=default classic script fetch options=].
+
+1. Let |base URL| be the [=API base URL=] of |environment settings|.
+
+1. Let |function script| be the result of [=create a classic script=] with |function declaration|,
+   |environment settings|, |base URL|, and |options|.
+
+1. [=Prepare to run function script=] with |environment settings|.
+
+1. Set |function evaluation status| to [=ScriptEvaluation=](|function script|'s record).
```

Comment:
Do we actually want the function to end up in the window global? e.g. if I send `functionDeclaration: "function foo() {}"`, should `window.foo` end up existing?

Reply:
Also, what happens if this doesn't result in a callable? For example if I pass in ``functionDeclaration: "2"`? Or something with side effects like `functionDeclaration: "window.foo` = 1"`?

Reply:
> Do we actually want the function to end up in the window global

No! I'm still working on step definitions, and hope to fix it soon.

Reply:
What does CDP do here?